### PR TITLE
Bug 844717 - Add Thai input method for Gaia. R=gpgreen@gmail.com

### PR DESCRIPTION
**Pull request by gpgreen**
Incorporated review comments from Rudy Lu.

### CODE REVIEW

File: apps/keyboard/js/layouts/th.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  shortLabel: 'Th',
+  menuLabel: 'ไทย', /*ไทย*/
+  basicLayoutKey: 'กขค', /*กขค*/
+  imEngine: 'latin',
+  types: ['text', 'url', 'email'],
+  pages: [ {                    // default page
+    width: 11,
+    secondLayout: true,
+    alt: {
+      'ช': '๏๚๛'
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: Change this to an array form.

File: apps/keyboard/js/layouts/th.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    keys: [
+      [
+        { value: 'ๅ' }, { value: 'ฃ' }, { value: 'ภ' }, { value: 'ถ' }, 
+        { value: 'ุ' }, { value: 'ึ' }, { value: 'ค' }, { value: 'ต' },
+        { value: 'จ' }, { value: 'ข' },
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: Please help remove the extra space at the end of this line.

File: apps/keyboard/js/layouts/th.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+Keyboards.th = {
+  label: 'Thai',
+  shortLabel: 'Th',
+  menuLabel: 'ไทย', /*ไทย*/
+  basicLayoutKey: 'กขค', /*กขค*/
+  imEngine: 'latin',
```
Comment (RudyLu) [inline apps/keyboard/js/layouts/th.js]: I suppose this won't need latin IME engine?

Comment (gpgreen) [inline apps/keyboard/js/layouts/th.js]: Yes, you are correct. Thai is not a latin language. I had left it there from the previous patch. Looking through the code I see that there is a default IME engine, which is fine for this.

**Comment by try-server-hook** [comment]
[gpgreen Greg Green (gpgreen)](https://github.com/gpgreen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=3c24b70d1b7b)

**Comment by gpgreen** [comment]
Rudy,

I have fixed the things you pointed out. This is my first change to Mozilla, so please stick with me. I am learning a lot about the review process here...

**Comment by try-server-hook** [comment]
[gpgreen Greg Green (gpgreen)](https://github.com/gpgreen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=968a6200756c)

**Comment by try-server-hook** [comment]
[gpgreen Greg Green (gpgreen)](https://github.com/gpgreen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b62b34a472dd)

**Pull request merged**